WasatchVCPPLib/WasatchVCPPLib/WasatchVCPPWrapper.cpp

```diff
@@ -765,57 +764,46 @@ int wp_get_raman_intensity_factors(int specIndex, double* factors, int factorsLe
     auto spec = driver->getSpectrometer(specIndex);
     if (spec == nullptr)
         return WP_ERROR_INVALID_SPECTROMETER;
-    double* out_factor; 
+
     auto eeprom = spec->eeprom;
-    float logTen;
-    float x_to_i;
-    float scaled;
-    float expanded;
-    int i;
-    int j;
+    if (eeprom.intensityCorrectionCoeffs.empty())
+        return WP_ERROR_NO_CALIBRATION;
     
     // expanding coeffs and calculating factors
-    if (!eeprom.intensityCorrectionCoeffs.empty())
+    for (int i = 0; i < factorsLen; i++)

```

Comment:
Keep iteration logic in for-loop.
```
int ROIHorizEnd = eeprom.ROIHorizStart + factorsLen;
for (int i = eeprom.ROIHorizStart; i < ROIHorizEnd; i++)
```

You could rename `i` to `px`, but we don't need both variables. If you rename you should change `x_to_i` to `x_to_px`

Reply:
If we did that, then we'd be doing this in the end, which doesn't feel like an improvement. Having to constantly juggle offsets without clearly labeled indexes is what led to the original code being buggy.
```
factors[i - eeprom.ROIHorizStart] = pow(10, logTen);
```

Reply:
You're right I didn't see Ln783